homework/index.js

```diff
@@ -29,23 +31,136 @@
     return elem;
   }
 
+  function createElement(ul, element1, element2, options1, options2) {
+    const li = createAndAppend('li', ul, { class: 'list' });
+    createAndAppend(element1, li, options1);
+    createAndAppend(element2, li, options2);
+  }
+
+  function dateChange(date) {
+    return new Date(date).toLocaleString();
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    ul.innerHTML = ' ';
+    createElement(
+      ul,
+      'h3',
+      'a',
+      { text: ' Repository:  ' },
+      {
+        text: repo.name,
+        href: repo.html_url,
+        target: '_blank',
+      },
+    );
+    createElement(
+      ul,
+      'h3',
+      'span',
+      { text: ' Description:  ' },
+      { text: repo.description },
+    );
+    createElement(
+      ul,
+      'h3',
+      'span',
+      { text: ' Forks:  ' },
+      { text: repo.forks },
+    );
+    createElement(
+      ul,
+      'h3',
+      'span',
+      { text: ' Updated:  ' },
+      { text: dateChange(repo.updated_at) },
+    );
+  }
+
+  function renderContributes(repositories, ul) {
+    const url = repositories.contributors_url;
+    fetchJSON(url)
+      .then(contributions => {
+        ul.innerHTML = ' ';
+        contributions.forEach(contributor => {
+          const li = createAndAppend('li', ul);
+          const infoImg = createAndAppend('div', li, { class: 'infoImg' });
+          createAndAppend('img', infoImg, {
+            src: contributor.avatar_url,
+            alt: contributor.login,
+          });
+          const infoName = createAndAppend('div', li, { class: 'infoName' });
+          createAndAppend('a', infoName, {
+            text: contributor.login,
+            href: contributor.html_url,
+            target: '_blank',
+          });
+          createAndAppend('span', infoName, {
+            text: contributor.contributions,
+            class: 'number',
+          });
+        });
+      })
+      .catch(err => {
+        createAndAppend('div', root, {
```

Comment:
The variable `root` is undefined here. This produces an ESLint error, causing the Travis check on your PR to fail. You can use `getElementById('root')` to avoid this.

```js
.catch(err => {
  createAndAppend('div', document.getElementById('root'), {
    text: err.message,
    class: 'alert-error',
  });
});
```